specification/network/resource-manager/Microsoft.Network/stable/2022-01-01/cloudServiceVipSwap.json

```diff
@@ -0,0 +1,180 @@
+          },
+          {
+            "$ref": "./network.json#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "./network.json#/definitions/CloudError"
+            }
+          },
+          "200": {
+            "description": "Request successful. The operation returns the slot type for the specified cloud service.",
+            "schema": {
+              "$ref": "#/definitions/SwapResource"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Get swap resource": {
+            "$ref": "./examples/GetSwapResource.json"
+          }
+        }
+      },
+      "put": {
+        "tags": [
+          "VipSwap"
+        ],
+        "operationId": "VipSwap_Create",
+        "description": "Performs vip swap operation on swappable cloud services.",
+        "parameters": [
+          {
+            "name": "groupName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the resource group."
+          },
+          {
+            "name": "resourceName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the cloud service."
+          },
+          {
+            "name": "singletonResource",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "Specifies a synthetic resource that cannot be created or deleted. Value must be set to default."
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/SwapResource"
+            },
+            "description": "SwapResource object where slot type should be the target slot after vip swap for the specified cloud service."
+          },
+          {
+            "$ref": "./network.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "./network.json#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Request successful."
+          },
+          "202": {
+            "description": "Accepted and the operation will complete asynchronously."
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "./network.json#/definitions/CloudError"
+            }
+          }
+        },
+        "x-ms-long-running-operation": true,
+        "x-ms-examples": {
+          "Put vip swap operation.": {
+            "$ref": "./examples/CreateSwapResource.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "SwapResource": {
```

Comment:
this is being exposed as a resource and should contain top level proxy resource properties (id, name, and type)

Reply:
SwapResource is a dummy resource in the NRP code and its not actually inheriting any topLevel resource. Even if I add these properties (id, name and type), they wont get exercised in the code anywhere. These properties does not make sense for SwapResource IMO. Do I still need to add them ?

Reply:
Customers (and other internal services0 expect APIs that look like resources to have standard properties.  The RPC defines the singleton pattern (and what resources look like).  https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/common-api-contracts.md#singleton-resources